src/test/java/BaseTest.java

```diff
@@ -1,13 +1,132 @@
-import org.testng.annotations.BeforeSuite;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.*;
+import java.time.Duration;
 
 public class BaseTest {
+    static WebDriver driver;
+    static String url;
+    static WebDriverWait wait;
+   public static void browserConfigs() {
 
+        driver.get(url);
+        url = "https://bbb.testpro.io/";
+        driver.manage().window().maximize();
+        provideEmail("dcabdi@gmail.com");
+        providePassword("te$t$tudent");
+    }
+    @DataProvider(name = "InvalidCredentials")
+    public static Object [][] getDataFromDataProviders(){
+        return new Object[][] {
+                {"invalid@class.com", "invalidPass"},
+                {"d@clas.com", ""},
+                {"" , ""}
+        };
+    }
+    protected static void playlistWasDeleted() throws InterruptedException {
+
+        WebElement deletedAlert = driver.findElement(By.xpath("//div[@class='alertify-logs top right']"));
+        Assert.assertTrue(deletedAlert.isDisplayed());
+    }
+    protected static void playlistDeleteBtn() throws InterruptedException {
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[@title='Delete this playlist']"))).click();
+    }
+    protected static void playlistToDelete() throws InterruptedException {
+
+       WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+       wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//a[@href='#!/playlist/26571']"))).click();
+    }
```

Comment:
Here is an example using the method `waitForElementToBeClickable(WebElement element)`

 ```
protected static void playlistDeleteBtn() {
        WebElement buttonDelete = driver.findElement(By.xpath("//button[@title='Delete this playlist'"));
        waitForElementToBeClickable(buttonDelete).click();
    }
```
![image](https://user-images.githubusercontent.com/104761482/198939672-e8f0e996-e131-439e-9333-2cd93afe4dd0.png)